roxie/ccd/ccdfile.cpp

```diff
@@ -1839,12 +1791,29 @@ class CRoxieFileCache : implements IRoxieFileCache, implements ICopyFileProgress
         if (!dllserver_root)
             return;
 #endif
+        unsigned cacheWarmTraceLevel = topology->getPropInt("@cacheWarmTraceLevel", traceLevel);
         VStringBuffer cacheFileName("%s/%s/cacheInfo.%d", dllserver_root, roxieName.str(), channel);
         StringBuffer cacheInfo;
         try
         {
             if (checkFileExists(cacheFileName))
             {
+#ifndef _WIN32
+                StringBuffer output;
+                VStringBuffer command("ccdcache %s -t %u", cacheFileName.str(), cacheWarmTraceLevel);
+                unsigned retcode = runExternalCommand(nullptr, output, output, command, nullptr);
+                if (output.length())
+                {
+                    StringArray outputLines;
+                    outputLines.appendList(output, "\n");
+                    ForEachItemIn(idx, outputLines)
+                    {
+                        DBGLOG("ccdcache: %s", outputLines.item(idx));
+                    }
+                }
+                if (retcode)
+                    DBGLOG("ccdcache failed with exit code %u", retcode);
+#endif
                 cacheInfo.loadFile(cacheFileName, false);
                 warmOsCache(cacheInfo);
```

Comment:
found this a little confusing.
It's now calling warmOsCache via ccdcache to warm disk cache, and calling it again here in Roxie to pre-load indexes (different code via presence of _STANDALONE_CCDCACHE)
Could it logically be split into 2 separate functions?

Reply:
I kept the code common because I wanted to make sure the parsing of the info was in sync between the two cases. It could possibly be done as a parse with a callback or lambda or something perhaps, but not sure if worth it